### DISCUSSION TITLE
Markdown to Typst escapes #

### DISCUSSION THREAD

**Discussion by kravlost** [jgm/pandoc#10316] [Announcements]
I want to add a page break by writing `#pagebreak()` in Markdown, but if I convert it to Typst or use Typst to convert the Markdown to PDF, then the # is escaped to \\# and Typst ignores it.

Is there a way of getting pandoc to transfer a # to Typst without escaping it so I can write Typst functions in Markdown?

Pandoc 3.5 on Windows 11.

**Comment by jgm** [comment]
Yes, you can use the `raw_attribute` extension (enabled by default in pandoc's markdown, and can be enabled in `gfm` using `-t gfm+raw_attribute`):
```
`#pagebreak()`{=typst}
```

**Reply by kravlost** [reply]
Genius, thanks!

**Comment by iandol** [comment]
@kravlost -- this excellent Lua filter supports Typst and allows you to use a generic `\pagebreak` or `\newline`, it makes your markdown easier to use with multiple final formats:

https://github.com/pandoc-ext/pagebreak

```
> pandoc -L pagebreak.lua -t typst

# Title

Some text.

\pagebreak

# Title

More text.


= Title
<title>
Some text.

#pagebreak()

= Title
<title-1>
More text.
```

**Comment by kravlost** [comment]
@jgm   If I add a Typst page setup block at the top of a markdown file introduced with 

```
```{=typst}
```

and complile to PDF via typst, there is an extra blank page at the top of the PDF. If I export to Typst and then complile it through Typst, there is no extra page.

Minimum example saved to `min-eg.md`:

```
```{=typst}
#set page(
  paper: "a4",
  margin: (x: 2cm, y: 2cm),
)

#align(
  horizon + center,
  [

    #set text(size: 20pt)

    *Minimum example*
  ]
)


#set page(numbering: "1")

#set heading(numbering: "1.1")


#show heading: it => {
  v(1em)
  it
  v(1em)
}
```{comment to show closing backticks on github, this shouldn't be here}


# First section

Goes here
```

Compliled with: `pandoc --pdf-engine=typst -f markdown -o min-eg.pdf min-eg.md` this shows an empty page before the first text.

**Comment by jgm** [comment]
I'm seeing the blank page in both cases: direct to pdf with pandoc or produce standalone typst and then compile with typst.

**Comment by jgm** [comment]
Note that the blank page is the title page.  If you specify a title, it will not be blank.

Instead of adding this command to set the page parameters, why not use the variables already encoded in the default template?

```
---
margin:
- x: 2cm
- y: 2cm
paper: a4
...
```
or set them in a defaults file.

**Reply by kravlost** [reply]
@jgm If I do this, I get an error when compiling to PDF:

```
error: expected string, found integer
    ┌─ \\?\C:\Users\.....\toPB079.html:129:11
    │
129 │   margin: (1: true,2: true,),
    │            ^

error: expected string, found integer
    ┌─ \\?\C:\Users\.......\toPB079.html:129:19
    │
129 │   margin: (1: true,2: true,),
    │                    ^

Error producing PDF.
```

**Reply by kravlost** [reply]
It works with:

```
margin:
  x: 2cm
  y: 2cm
```

**Reply by jgm** [reply]
Yeah, sorry -- that's what I meant  to write!

**Reply by kravlost** [reply]
It's working well now with the title page data in the yaml block, the only thing I'm having to do is add a page break function in Typst so the title data is on a page of its own (pdflatex does this itself).

Thanks for all the help!

**Comment by avidseeker** [comment]
I wonder if it's possible for typst syntax to be mixed with markdown? Would it be possible to have a `raw_typst` extension similar to `raw_tex` or `raw_html`?

**Reply by iandol** [reply]
With some specific parsing rules you can use a Lua filter, for example I have a filter so I can use native Typst cross-reference labels and references directly in my markdown: https://github.com/iandol/dotpandoc/blob/master/filters/typstFix.lua — you could build a filter like that to parse more native Typst…
